sale_promotion_discount_in_field/models/sale_order.py

```diff
@@ -0,0 +1,131 @@
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _create_new_no_code_promo_reward_lines(self):
+        """
+        Override core method to save program with new discount apply rule
+        """
+        self.ensure_one()
+        order = self
```

Comment:
can you make this a bit more readable? Maybe you can save some indentation and split this method into smaller pieces?